### PR TITLE
Add support for visionOS and macOS

### LINKED ISSUE #19: Text input is very slow on visionOS.
### Description

The current `enter(...)` method for text entry is really slow on visionOS.

### Reproduction

Just try to enter text into a text field.

### Expected behavior

Have it faster :)

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

### PR DESCRIPTION
**Pull request by Supereg**
# Optimize text input on visionOS

## :recycle: Current situation & Problem
The current approach for textfield selection does not work on visionOS. This PR addresses the issue by replacing the current solution via a simple "tap" on visionOS. Our automatic text input correction should catch the case where the tap will set the cursor in the middle of the text.

This resolves #19.

Further, keyboards and alerts are in separate processes in visionOS. This PR accounts for that and updates the respective implementations.

## :gear: Release Notes 
* Fix smaller issues with visionOS.

## :books: Documentation
Minor adjustments.

## :white_check_mark: Testing
* Added a new test case that explicitly checks the use case of entering text in a text field, prefilled with long text.
* Added CI support for visionOS.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
Would require an updated branch protection 👍

**Comment by Supereg** [comment]
Couple issues here with the visionOS builds. `com.apple.springboard` seems to not exists there. Also the keyboard is launched as a separate app and we can't access it with the same application. There is probably a way to access it via UI tests but I'm not sure how.

**Comment by Supereg** [comment]
@PSchmiedmayer this PR is ready to review and merge. The branch protection rules would need an update due to changes to the job names.

File: Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift [hunk 1]
```diff
@@ -10,14 +10,37 @@ import XCTest
...
+    /// The bundle identifier for the home screen for the specific platform.
+    ///
+    /// E.g., `com.apple.springboard` on iOS.
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    public static var homeScreenBundle: String {
+#if os(visionOS)
+        "com.apple.RealityLauncher"
+#elseif os(tvOS)
+        "com.apple.pineboard"
```
Comment (PSchmiedmayer) [inline Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift:22]: 🌲

File: Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift [hunk 2]
```diff
@@ -10,14 +10,37 @@ import XCTest
...
 
 extension XCUIApplication {
+    /// The bundle identifier for the home screen for the specific platform.
+    ///
+    /// E.g., `com.apple.springboard` on iOS.
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    public static var homeScreenBundle: String {
+#if os(visionOS)
+        "com.apple.RealityLauncher"
```
Comment (PSchmiedmayer) [inline Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift:20]: 👓

File: Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift [hunk 3]
```diff
@@ -10,14 +10,37 @@ import XCTest
...
+#elseif os(tvOS)
+        "com.apple.pineboard"
+#elseif os(iOS)
+        "com.apple.springboard"
```
Comment (PSchmiedmayer) [inline Sources/XCTestExtensions/XCUIApplication+DeleteAndLaunch.swift:24]: 🌱

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the additions 🎉

**Pull request merged**